subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/verification/DependencyVerificationSignatureCheckIntegTest.groovy

```diff
@@ -528,4 +528,30 @@ This can indicate that a dependency has been compromised. Please verify carefull
   - Artifact foo-1.0.jar (org:foo:1.0) checksum is missing from verification metadata."""
     }
 
+    def "unsigned artifacts require checksum verification"() {
+        createMetadataFile {
+            keyServer(keyServerFixture.uri)
+            verifySignatures()
+            addChecksum("org:foo:1.0", "sha256", "nope")
+            addChecksum("org:foo:1.0", "sha256", "nope", "pom", "pom")
+        }
+
+        given:
+        javaLibrary()
+        uncheckedModule("org", "foo", "1.0")
+        buildFile << """
+            dependencies {
+                implementation "org:foo:1.0"
+            }
+        """
+
+        when:
+        fails ":compileJava"
+
+        then:
+        failure.assertHasCause """Dependency verification failed for configuration ':compileClasspath':
+  - On artifact foo-1.0.jar (org:foo:1.0): expected a 'sha256' checksum of 'nope' but was '20ae575ede776e5e06ee6b168652d11ee23069e92de110fdec13fbeaa5cf3bbc'
```

Comment:
How complicated would it be to clearly state we attempted a signature check but could not.
Which is different than having a module configured to ignore signature, or not having signature check enabled at all.
Makes sense?

Reply:
Yes, I'll try to improve this.

Reply:
Done